### ISSUE TITLE
ASYNC115 and ASYNC116 check the wrong argument name for `anyio.sleep`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#18164]
### Summary

[`async-zero-sleep` (ASYNC115)](https://docs.astral.sh/ruff/rules/async-zero-sleep/) and [`long-sleep-not-forever` (ASYNC116)](https://docs.astral.sh/ruff/rules/long-sleep-not-forever/) check both `trio.sleep` and `anyio.sleep`. The argument can be specified by position or name. Trio names it `seconds` whereas AnyIO names it `delay`. ASYNC115 and ASYNC116 always check for the name `seconds`, causing false positives and false negatives for AnyIO.

```console
$ cat <<'# EOF' | ruff --isolated check - --preview --select ASYNC115,ASYNC116 --unsafe-fixes --diff -q
import anyio
async def async115_false_positive(): await anyio.sleep(seconds=0)
async def async115_false_negative(): await anyio.sleep(delay=0)
async def async116_false_positive(): await anyio.sleep(seconds=86401)
async def async116_false_negative(): await anyio.sleep(delay=86401)
# EOF
@@ -1,5 +1,5 @@
 import anyio
-async def async115_false_positive(): await anyio.sleep(seconds=0)
+async def async115_false_positive(): await anyio.lowlevel.checkpoint()
 async def async115_false_negative(): await anyio.sleep(delay=0)
-async def async116_false_positive(): await anyio.sleep(seconds=86401)
+async def async116_false_positive(): await anyio.sleep_forever()
 async def async116_false_negative(): await anyio.sleep(delay=86401)


```

### Version

ruff 0.11.10 (b35bf8ae0 2025-05-15)

**Comment by Vasanth-96** [comment]
Hi I would like to contribute to this issue, thanks
